### PR TITLE
clustered index support IndexScan

### PR DESCRIPTION
**Pull request by marsishandsome**
### What is changed and how it works?
support IndexScan on `clustered index table` (primary key has only one column)

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/codec/CodecDataInput.java [hunk 1]
```diff
@@ -206,30 +206,36 @@ public int cutOne() {
...
-        //        l = peekCompactBytes(b);
-        //      case DECIMAL_FLAG:
-        //        l = types.DecimalPeak(b);
-        //      case VARINT_FLAG:
-        //        l = peekVarint(b);
-        //      case UVARINT_FLAG:
+      case DECIMAL_FLAG:
+        DecimalCodec.readDecimal(this);
+        break;
+        // case VARINT_FLAG:
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/codec/CodecDataInput.java:228]: is this case possible？

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java [hunk 2]
```diff
@@ -303,7 +303,13 @@ private DAGRequest buildRequest(
         // double read case
         if (!hasPk) {
           // add handle column
-          indexScanBuilder.addColumns(handleColumn);
+          if (!tableInfo.isCommonHandle()) {
+            indexScanBuilder.addColumns(handleColumn);
+          } else {
+            for (TiIndexColumn col : tableInfo.getPrimaryKey().getIndexColumns()) {
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java:309]: i think haspk is false？

File: tikv-client/src/main/java/com/pingcap/tikv/codec/CodecDataInput.java (hunk 1 above)
Comment (marsishandsome) [reply tikv-client/src/main/java/com/pingcap/tikv/codec/CodecDataInput.java:228]: not occurred in test

**Review by birdstorm — APPROVED** [review]
LGTM

**Pull request merged**